### PR TITLE
Add optional help text to contentanalysis

### LINKED ISSUE #476: Add optional help text to the ContentAnalysis component
Add an optional help text to the ContentAnalysis component
38875653-21b37e2a-425b-11e8-8974-91c6245679ff

- The help text should be optional.
- The separator should only be there when there is also a help text.

**please branch from feature/content-analysis-in-sidebar**

### PR DESCRIPTION
**Pull request by igorschoester**
## Summary

This PR can be summarized in the following changelog entry:

* Added a help text component.

## Relevant technical choices:

* Uses PropType one of to make it possible to pass the text in an array as well as just a string. In the array you can pass a link for example. This prop type is exported to be used in a component that uses HelpText.
* Uses a paragraph element with a default style as main parent.

## Test instructions

This PR can be tested by following these steps:

* View the HelpText in action: `yarn start` -> `localhost:3333` -> tab `Content analysis`.
* Check `yarn test` && `grunt check`.

Fixes #476

### CODE REVIEW

File: composites/Plugin/ContentAnalysis/components/ContentAnalysis.js [hunk 1]
```diff
@@ -205,21 +208,25 @@ class ContentAnalysis extends React.Component {
...
+		const errorsFound = errorsResults.length;
+		const problemsFound = problemsResults.length;
+		const improvementsFound = improvementsResults.length;
+		const considerationsFound = considerationsResults.length;
+		const goodResultsFound = goodResults.length;
 
 		// Analysis collapsibles are only rendered when there is at least one analysis result for that category present.
 		return (
 			<ContentAnalysisContainer>
+				{ helpText && <HelpText text={helpText} /> }
```
Comment (afercia) [inline composites/Plugin/ContentAnalysis/components/ContentAnalysis.js]: Please add spaces within the brackets.

File: composites/Plugin/ContentAnalysis/tests/ContentAnalysisTest.js [hunk 2]
```diff
@@ -53,199 +53,223 @@ const considerationsResults = [
...
+			<ContentAnalysis
+				problemsResults={ problemsResults }
+				improvementsResults={ improvementsResults }
+				goodResults={ goodResults }
+				considerationsResults={ considerationsResults }
+				errorsResults={ errorsResults }
+				changeLanguageLink={ "#" }
+				language="English"
+				showLanguageNotice={ true }
+				canChangeLanguage={ true }
```
Comment (afercia) [inline composites/Plugin/ContentAnalysis/tests/ContentAnalysisTest.js]: Shouldn't this be false? it's testing for "someone who cannot change the language"

File: composites/Plugin/Shared/components/HelpText.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/* External dependencies */
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+/* Internal dependencies */
+import colors from "../../../../style-guide/colors";
+
+const YoastHelpText = styled.p`
+	color: ${colors.$color_grey_text};
```
Comment (afercia) [inline composites/Plugin/Shared/components/HelpText.js]: Please add spaces within brackets.

File: composites/Plugin/Shared/components/HelpText.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	 * Renders a help text component.
+	 *
+	 * @returns {ReactElement} The rendered help text component.
+	 */
+	render() {
+		const { text } = this.props;
+
+		return (
+			<YoastHelpText>
+				{text}
```
Comment (afercia) [inline composites/Plugin/Shared/components/HelpText.js]: Please add spaces within brackets.

**Comment by afercia** [comment]
CR 🚧 

A few minor coding standards and one typo in the tests.

File: composites/Plugin/ContentAnalysis/tests/ContentAnalysisTest.js (hunk 2 above)
Comment (igorschoester) [reply composites/Plugin/ContentAnalysis/tests/ContentAnalysisTest.js]: Nice spot. That makes complete sense to me yes. Changing.

**Comment by danielFangstrom** [comment]
CR 👍

**Comment by danielFangstrom** [comment]
ACC 👍

**Pull request merged**